test/box/broken_snapshot.test.lua

```diff
@@ -0,0 +1,123 @@
+
+-- write data recover from latest snapshot
+env = require('test_run')
+test_run = env.new()
+
+test_run:cmd("restart server default")
+fio = require 'fio'
+test_run:cmd("setopt delimiter ';'")
+function get_snap_file ()
+    local snapfile = nil	
+    local directory = fio.pathjoin(fio.cwd(), 'broken_snapshot')
+    for files in io.popen(string.format("ls %s", directory)):lines() do
+        local snaps = string.find(files, "snap")
+        if (snaps ~= nil) then
+            local snap = string.find(files, "%n")
+	    if (snap ~= nil) then
+                snapfile = string.format("%s/%s", directory, files)
+            end
+        end
+    end
+    return snapfile
+end;
+test_run:cmd("setopt delimiter ''");
+
+
+test_run:cmd("create server test with script='box/broken_snapshot.lua'")
+test_run:cmd("start server test")
+test_run:cmd("switch test")
+space = box.schema.space.create('test', { engine = "memtx" })
+space:format({ {name = 'id', type = 'unsigned'}, {name = 'year', type = 'unsigned'} })
+index = space:create_index('primary', { parts = {'id'} })
+
+for key = 1, 10000 do space:insert({key, key + 1000}) end
+box.snapshot()
```

Comment:
Instead of generating .snap file on each test run, mb it is worth creating it once and put it near test? E.g. see upgrage tests. It would speed up test a little.